### PR TITLE
[r] Proper prefixing for shape-related methods

### PR DESCRIPTION
**Pull request by johnkerl**
**Issue and/or context:** As tracked on issue #2407 / [[sc-51048]](https://app.shortcut.com/tiledb-inc/story/51048).

Note that the intended Python and R API changes are all agreed on and finalized as described in #2407.

**Changes:**

Direct R analogue of #3236 

**Notes for Reviewer:**

### CODE REVIEW

**Review by mojaveazure — CHANGES_REQUESTED** [review]

File: apis/r/R/SOMADataFrame.R [hunk 1]
```diff
@@ -415,13 +415,15 @@ SOMADataFrame <- R6::R6Class(
...
+    tiledbsoma_resize_soma_joinid_shape = function(new_shape) {
 
       stopifnot("'new_shape' must be an integer" = rlang::is_integerish(new_shape, n = 1) ||
         (bit64::is.integer64(new_shape) && length(new_shape) == 1)
       )
       # Checking slotwise new shape >= old shape, and <= max_shape, is already done in libtiledbsoma
-      invisible(resize_soma_joinid_shape(self$uri, new_shape, private$.soma_context))
+      invisible(
+        resize_soma_joinid_shape(
+          self$uri, new_shape, "tiledbsoma_resize_soma_joinid_shape", private$.soma_context))
```
[the remark this answers is not in the document]
Comment (johnkerl) [reply apis/r/R/SOMADataFrame.R:426]: Please see the discussion here:

https://github.com/single-cell-data/TileDB-SOMA/pull/3236#issuecomment-2438000530

These are absolutely crucial as narrated.

What I'll do on this PR (on my plan for today, sorry to not have put this PR back in draft until I did so) is to use this pattern on the R side:

https://github.com/single-cell-data/TileDB-SOMA/pull/3236#issuecomment-2438078356

[the remark this answers is not in the document]
Comment (johnkerl) [reply apis/r/R/SOMADataFrame.R]: @mojaveazure https://github.com/single-cell-data/TileDB-SOMA/pull/3237/commits/b84a478ffe32802727a41ed7ebda1dd5046ccb45

File: apis/r/R/SOMADataFrame.R [hunk 2]
```diff
@@ -415,13 +415,16 @@ SOMADataFrame <- R6::R6Class(
     #' @param new_shape An integer, greater than or equal to 1 + the
     #' `soma_joinid` domain slot.
     #' @return No return value
-    resize_soma_joinid_shape = function(new_shape) {
+    tiledbsoma_resize_soma_joinid_shape = function(new_shape) {
 
       stopifnot("'new_shape' must be an integer" = rlang::is_integerish(new_shape, n = 1) ||
         (bit64::is.integer64(new_shape) && length(new_shape) == 1)
       )
+      function_name_for_messages <- sys.call(sys.parent(n=1)[[1]]
```
Comment (mojaveazure) [inline apis/r/R/SOMADataFrame.R:423]: This is incorrect
```suggestion
      function_name_for_messages <- as.character(sys.call())
```
`sys.parent(1L)` would be the parent of the method, not the method itself (eg. `R_GlobalEnv` or the testthat `eval` environment); `sys.call()` also returns a `call`, and subsetting a `call` does not always return a `character`

Note, this returns `<name-of-object>$<method_name>`; if you want just the method name, we would need
```R
sub('.*\\$', replacement = '', x = as.character(sys.call()))
```

**Review by mojaveazure — CHANGES_REQUESTED** [review]

**Review by mojaveazure — APPROVED** [review]
Please bump the develop version, update the changelog, and then 🚢

**Pull request merged**